app/controllers/ForumPost.scala

```diff
@@ -37,30 +38,49 @@ final class ForumPost(env: Env) extends LilaController(env) with ForumController
         if topic.closed then BadRequest("This topic is closed")
         else if topic.isOld then BadRequest("This topic is archived")
         else
-          categ.team.so(env.team.api.isLeader(_, me)) flatMap { inOwnTeam =>
-            forms
-              .post(inOwnTeam)
-              .bindFromRequest()
-              .fold(
-                err =>
-                  CategGrantWrite(categId, tryingToPostAsMod = true):
-                    for
-                      captcha     <- forms.anyCaptcha
-                      unsub       <- env.timeline.status(s"forum:${topic.id}")
-                      canModCateg <- access.isGrantedMod(categ.slug)
-                      page <- renderPage:
-                        html.forum.topic
-                          .show(categ, topic, posts, Some(err -> captcha), unsub, canModCateg = canModCateg)
-                    yield BadRequest(page)
-                ,
-                data =>
-                  CategGrantWrite(categId, tryingToPostAsMod = ~data.modIcon):
-                    CreateRateLimit(ctx.ip, rateLimited):
-                      postApi.makePost(categ, topic, data) map { post =>
-                        Redirect(routes.ForumPost.redirect(post.id))
-                      }
-              )
-          }
+          for
+            canModCateg <- access.isGrantedMod(categ.slug)
+            topicUserId = topic.userId.getOrElse(UserId(""))
+            relation <- ctx.userId.so(
+              env.relation.api.fetchRelation(topicUserId: UserId, _)
+            )
+            isUblog      = topic.ublogId.isDefined
+            replyBlocked = relation.exists(_ == Block) && !canModCateg && isUblog
```

Comment:
`relation.contains(Block)`

Reply:
if we only need the relation when `isUblog`, then we should only fetch the relation from database when `isUblog`.